lti_consumer/models.py

```diff
@@ -132,8 +139,20 @@ def _get_lti_1p3_consumer(self):
 
             # Check if enabled and setup LTI-AGS
             if self.block.has_score:
+
+                # create LineItem if there is none for current lti configuration
+                lineitem, _ = LtiAgsLineItem.objects.get_or_create(
```

Comment:
This is missing a few parameters. Check https://github.com/edx/xblock-lti-consumer/blob/master/docs/decisions/0003-lti-1p3-score-linking.rst#declarative-grade-handling for details on all attributes.

Reply:
I've added `lti_configuration`, `resource_id`, `label`, `score_maximum` (`weight`). I didn't find others in attributes. Let me know if I've missed.

Reply:
There's `start_date_time` and `end_date_time` which might be available on some cases.
You can check if the problem is past due by looking at https://github.com/edx/xblock-lti-consumer/blob/bcdcf218096d63b88a8d8c56e027c0910814c381/lti_consumer/lti_xblock.py#L828

Reply:
@shimulch `resource_link_id` is missing and should be set to the problem's location.

Reply:
@giovannicimolin I've put ``resource_link_id`` as same as ``self.block.location``. Also checked if ``start`` and ``due`` is set. If set then saved as ``start_date_time`` and ``end_date_time`` accordingly.